### PR TITLE
added Support for NO_BOOST

### PR DESCRIPTION
**Pull request by sebt3**
Same goal as previous PR : Remove Boost dependency.
Change compared to previous PR :
- define namespace instead of C macros
- using boost keep being the default
- no cmake changes
- define NO_BOOST to have the expected behaviour

### CODE REVIEW

**Comment by sebt3** [comment]
Trying to fix #122 here ;)

File: client_http.hpp [hunk 1]
```diff
@@ -30,6 +76,8 @@ class case_insensitive_hash {
     return seed;
   }
 };
```
Comment (eidheim) [inline client_http.hpp]: To make things simpler, we should maybe rewrite `case_insensitive_equals` and `case_insensitive_hash` to use std instead (even when using boost), and thus only have one definition of these two classes. I think `case_insensitive_hash` can use `std::hash` but have not looked into this in detail yet. Found this though after a quick search: https://stackoverflow.com/questions/2590677/how-do-i-combine-hash-values-in-c0x

Comment (eidheim) [inline client_http.hpp]: I've looked a bit on this, and could do this work I think. Never mind the above comment then:)

**Comment by eidheim** [comment]
From a quick look this looks good! I'll do a more detailed look in a day or two and merge. I'll also fix the case_insensitive classes and make the test run on standalone asio as well as boost::asio. Thank you again!

**Comment by sebt3** [comment]
If you have a good idea for a better "case_insensitive" then all the better. One of the reason why I havent made the boost-less version the default is that I felt the boost version was nicer.
I'm glad this version have a chance to be merged. Feel free to ask fixes all you need ;)

**Comment by eidheim** [comment]
@sebt3 I did some changes that you can see in the above commit. Let me know what you think.

**Comment by sebt3** [comment]
Sound very good to me. A bit more intrusive in your code than what I would have done, but it's not my code :P and your changes make the code way more readable (I disliked those "_ns").

The fact that you added tests for this on travis is an added bonus.

Thanks a lot
(gosh : did not comment the right page :( )

**Pull request merged**